src/gui/qgsmaptoolcapture.cpp

```diff
@@ -1381,13 +1383,13 @@ void QgsMapToolCapture::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
     if ( digitizingFinished )
     {
       QgsGeometry g;
-      QgsCurve *curveToAdd = captureCurve()->clone();
+      std::unique_ptr<QgsCurve> curveToAdd( captureCurve()->clone() );
 
       if ( mode() == CaptureLine )
       {
-        g = QgsGeometry( curveToAdd );
+        g = QgsGeometry( curveToAdd.get() );
```

Comment:
QgsGeometry takes ownership of curvetoadd so you have to release the unique_ptr.

Reply:
Actually you can just use std::move here, QgsGeometry has a constructor which accepts unique_ptr

Reply:
Indeed, but curveToAdd would be in an invalid state when firing the lineCaptured signal

Reply:
Std::move automatically nulls curveToAdd, just like .release would

Reply:
Right! I thought any moved resource  was in an invalid state, but it seems the standard explicitly describe what happen for unique_ptr. My bad.

But still, firing a signal with a null pointer seems an unwanted behavior, no?